echopop/live/live_data_loading.py

```diff
@@ -0,0 +1,928 @@
+    live_file_config_path = Path(live_file_config_path)
+    # ---- Create list of both config paths
+    config_files = [live_init_config_path, live_file_config_path]
+    # ---- List of file existence checks
+    config_existence = [live_init_config_path.exists(), live_file_config_path.exists()]
+    # ---- Error evaluation and print message (if applicable)
+    if not all(config_existence):
+        missing_config = [
+            files for files, exists in zip(config_files, config_existence) if not exists
+        ]
+        raise FileNotFoundError(
+            f"The following configuration files do not exist: {missing_config}."
+        )
+
+    # Read the YAML configuration/recipe file to parameterize the `LiveSurvey` class
+    # ---- Initialization settings
+    init_config = yaml.safe_load(Path(live_init_config_path).read_text())
+    # -------- Validate
+    init_config = validate_live_config(
+        copy.deepcopy(init_config), LIVE_CONFIG_INIT_MODEL, live_init_config_path
+    )
+    # ---- Filepath/directory settings
+    file_config = yaml.safe_load(Path(live_file_config_path).read_text())
+    file_config = validate_live_config(
+        copy.deepcopy(file_config), LIVE_CONFIG_DATA_MODEL, live_file_config_path
+    )
+
+    # Check for intersecting/duplicative configuration keys
+    # ---- Compare sets of keys from each dictionary
+    config_intersect = set(init_config.keys()).intersection(set(file_config.keys()))
+    # ---- Raise error if needed
+    if config_intersect:
+        raise ValueError(
+            f"The initialization and file configuration files comprise the following intersecting "
+            f"keys: {' ,'.join(config_intersect)}. Key names must be unique for each configuration "
+            f"file."
+        )
+
+    # Combine both into a dictionary output that can be added to the `LiveSurvey` class object
+    return {**init_config, **file_config}
+
+
+def read_acoustic_files(acoustic_files: List[str], xarray_kwargs: dict = {}) -> tuple:
+
+    # Get the file-specific settings, datatypes, columns, etc.
+    # ---- Get defined columns and datatypes from `LIVE_INPUT_FILE_CONFIG_MAP`
+    acoustics_config_map = LIVE_INPUT_FILE_CONFIG_MAP["acoustics"]
+
+    # Read all of the zarr files
+    results_list = [
+        (data_df, unit_dict) if i == 0 else (data_df, None)
+        for i, (data_df, unit_dict) in enumerate(
+            read_acoustic_zarr(file, acoustics_config_map, xarray_kwargs=xarray_kwargs)
+            for file in acoustic_files
+        )
+    ]
+
+    # Concatenate the dataframe component
+    acoustic_data_df = pd.concat([df for df, _ in results_list], ignore_index=True)
+    # ---- Add the `acoustic_data_units` to the dictionary and output the resulting tuple
+    return acoustic_data_df, results_list[0][1] if results_list else None
+
+
+def filter_filenames(
+    directory_path: Path, filename_id: str, files: List[Path], file_extension: str
+):
```

Comment:
I think we can use fsspec for flexible handling between different file systems. Also, it is a lot of work here to get something that is general across all the files, but the patterns are still pretty specific. For the scenario we are dealing with, I'd say generality is not the first priority, and more hard-coding may actually make the code more readable. Let's discuss this when we meet.